command/git_push.go

```diff
@@ -165,30 +171,33 @@ type pushParams struct {
 	authorEmail   string
 	commitMessage string
 	branch        string
+	needsCommit   bool
 }
 
 func (c *gitPush) pushPatch(ctx context.Context, logger client.LoggerProducer, p pushParams) error {
 	jpm := c.JasperManager()
 
-	author := fmt.Sprintf("%s <%s>", p.authorName, p.authorEmail)
-	commitCommand := fmt.Sprintf("git "+
-		`-c "user.name=%s" `+
-		`-c "user.email=%s" `+
-		`commit --file - `+
-		`--author="%s"`,
-		c.CommitterName, c.CommitterEmail, author)
-	logger.Execution().Debugf("git commit command: %s", commitCommand)
-	cmd := jpm.CreateCommand(ctx).Directory(p.directory).Append(commitCommand).SetInput(bytes.NewBufferString(p.commitMessage)).
-		SetOutputSender(level.Info, logger.Task().GetSender()).SetErrorSender(level.Error, logger.Task().GetSender())
-	if err := cmd.Run(ctx); err != nil {
-		return errors.Wrap(err, "can't create commit from files")
+	if p.needsCommit {
```

Comment:
Because that function re-reads the whole diff and requires us to re-grab it from the database I'm sort of against it. This check is only temporary for the migration also, because once it's done this entire chunk can be removed.

Reply:
I guess I didn't republish my reply here, but refactoring that function actually made this fine.